explorer/syntax/parser.ypp

```diff
@@ -446,10 +450,28 @@ multiplicative_expression:
           std::vector<Nonnull<Expression*>>({$1, $3}));
     }
 ;
+modulo_operand:
+  ref_deref_expression
+| additive_expression
+;
+modulo_lhs:
+  ref_deref_expression
+| modulo_expression
+;
+modulo_expression:
+  additive_expression
+| modulo_lhs MODULO modulo_operand
```

Comment:
```suggestion
modulo_operand:
  ref_deref_expression
| minus_expression
;
modulo_expression:
  modulo_operand MODULO modulo_operand
```

In Carbon, an unparenthesized additive expression isn't permitted as an operand of a modulo expression -- `a % b + c` requires parentheses, for example. Also, modulo expressions are not associative -- `a % b % c` requires parentheses. See the [precedence diagram](https://github.com/carbon-language/carbon-lang/tree/trunk/docs/design/expressions#precedence).

Reply:
Also, I wonder if we can give a better name to the `ref_deref_expression | minus_expression` rule that we now use as both `multiplicative_operand` and `modulo_operand`, and will be used a few more times for other operators too. Maybe `unary_expression`?

Reply:
Yes, I think you are reigh!t The best name is unary_expression...